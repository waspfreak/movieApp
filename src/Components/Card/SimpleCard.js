import React from 'react';
import {SimpleCardStyles} from './style';
import PropTypes from 'prop-types';

/**
 * Simple Card  UI component for user interaction
 */

export const SimpleCard = ({originalTitle, posterPath, id, ...props}) => {
	return (
		<SimpleCardStyles {...props} key={id}>
			<img className='image' src={posterPath} alt={originalTitle} />
		</SimpleCardStyles>
	);
};

SimpleCard.propTypes = {
	id: PropTypes.any,
	originalTitle: PropTypes.string,
	posterPath: PropTypes.string.isRequired,
};

export default SimpleCard;
