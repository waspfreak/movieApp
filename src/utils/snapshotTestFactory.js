import {cleanup, render} from '@testing-library/react';
import React from 'react';

/**
 *
 * Sets up snapshot tests in a standard way and avoids repeating boiler plate
 */
export function snapshotTestFactory(Component: any, properties: any) {
	/**
	 * Re-usable it block
	 */
	const itBlock = (props: any) => {
		it('should match snapshot', () => {
			const {container} = render(<Component {...props} />);
			expect(container).toMatchSnapshot();
		});
	};

	/**
	 * Re-usable describe block
	 */
	describe(`${Component.name}`, () => {
		/**
		 * Clean up any mock functions that might be in props
		 */
		afterEach(() => {
			jest.clearAllMocks();
			cleanup();
		});

		/**
		 * If its an array of props then iterate them into seperate it blocks
		 */
		if (Array.isArray(properties)) {
			properties.forEach((props) => {
				itBlock(props);
			});
		} else {
			itBlock(properties);
		}
	});
}
