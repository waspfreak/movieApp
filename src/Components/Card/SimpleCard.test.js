import {snapshotTestFactory} from '../../utils/snapshotTestFactory';
//
import {SimpleCard} from './SimpleCard';
const props: SimpleCard[] = [
	{
		posterPath:
			'http://image.tmdb.org/t/p/w500/lPsD10PP4rgUGiGR4CCXA6iY0QQ.jpg',
		originalTitle: 'Movie Title',
	},
];
snapshotTestFactory(SimpleCard, props);
