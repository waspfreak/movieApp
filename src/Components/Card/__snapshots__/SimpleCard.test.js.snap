// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleCard should match snapshot 1`] = `
.c0 {
  font-family: "Titillium Web",sans-serif;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-transition: -webkit-transform 0.45s;
  -webkit-transition: transform 0.45s;
  transition: transform 0.45s;
  width: 150px;
  margin-inline: 10px;
  margin-bottom: 20px;
  cursor: pointer;
}

.c0:hover {
  -webkit-transform: scale(1.03);
  -ms-transform: scale(1.03);
  transform: scale(1.03);
}

.c0 .image {
  width: 100%;
}

<div>
  <div
    class="c0"
  >
    <img
      alt="Movie Title"
      class="image"
      src="http://image.tmdb.org/t/p/w500/lPsD10PP4rgUGiGR4CCXA6iY0QQ.jpg"
    />
  </div>
</div>
`;
